jobs/keepalived/templates/keepalived_ctl

```diff
@@ -6,24 +6,34 @@ SBIN_DIR=${PKG_DIR}/sbin
 CONF_DIR=${APP_DIR}/config
 RUN_DIR=/var/vcap/sys/run/keepalived
 LOG_DIR=/var/vcap/sys/log/keepalived
-PIDFILE=${RUN_DIR}/pid
+
+exec >>${LOG_DIR}/keepalived.log 2>&1
 
 case $1 in
 
   start)
+    echo "Starting keepalived... ($(date))"
     mkdir -p $RUN_DIR $LOG_DIR
     chown -R vcap:vcap $RUN_DIR $LOG_DIR
-    echo $$ > $PIDFILE
-	$SBIN_DIR/keepalived \
-		-f $CONF_DIR/keepalived.config \
-		--pid=$PIDFILE \
-		&
-		
+    interface=$(ip route get 8.8.8.8 | tr -s "\n" " " | sed 's/.*\dev \(\S\+\).*/\1/')
+    if grep "interface auto" ${CONF_DIR}/keepalived.config > /dev/null; then
+        sed -i "s/interface auto/interface ${interface}/" ${CONF_DIR}/keepalived.config
+    fi
```

Comment:
Wouldn't this logic be better located in the erb template https://github.com/cloudfoundry-community/haproxy-boshrelease/blob/master/jobs/keepalived/templates/keepalived.config.erb#L21 possibly leveraging unit tested ruby method ?

Embedding the logic in the erb templates may have the benefit of leaving bosh director trigger a redeploy on incoming changes would have resulted in a changed rendered template.

However, it's not clear to me if the bosh supported interpolation variables would allow to predict the interface name on the director side.